actionview/lib/action_view/flows.rb

```diff
@@ -15,7 +15,7 @@ def get(key)
 
     # Called by each renderer object to set the layout contents.
     def set(key, value)
-      @content[key] = value
+      @content[key] = ActiveSupport::SafeBuffer.new(value)
```

Comment:
I believe the intention would be to make all values in the @content hash be instances of ActiveSupport::SafeBuffer, as implied by the constructor, which initializes all values of the @content hash to be ActiveSupport::SafeBuffer instances by default:

```ruby
def initialize
  @content = Hash.new { |h,k| h[k] = ActiveSupport::SafeBuffer.new }
end
```

Given this, it seems to me that the `append` method expects a new value to be appended to be an existing ActiveSupport::SafeBuffer object. A String object (not html_safe) appended to an ActiveSupport::SafeBuffer object will still result in an ActiveSupport::SafeBuffer object.

The `set` method as it currently stands is inconsistent with both the constructor and the `append` method. I believe this to be unintended, as it could wipe out the initial ActiveSupport::SafeBuffer object created in the constructor with a String.